config/src/config/state_sync_config.rs

```diff
@@ -233,6 +229,13 @@ pub struct DataStreamingServiceConfig {
     /// the subscription stream is terminated and a new stream must be created.
     pub max_num_consecutive_subscriptions: u64,
 
+    /// Maximum number of pending requests per data stream.
+    pub max_pending_requests: u64,
```

Comment:
SGTM! Thanks boss 😄 Added:

```
    /// Maximum number of pending requests per data stream. This includes the
    /// requests that have already succeeded but have not yet been consumed
    /// because they're head-of-line blocked by other requests.
```